mathml/relations/css-styling/first-line-first-letter-pseudo-elements-003.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>MathML and ::first-line/::first-letter pseudo-elements</title>
+<link rel="help" href="https://w3c.github.io/mathml-core/#legacy-mathml-style-attributes">
+<link rel="match" href="first-line-first-letter-pseudo-elements-003-ref.html"/>
+<meta name="assert" content="display math elements do not contribute a first formatted line/letter to ancestors.">
+<style>
+  .redfirstline::first-line { background: red; }
+  .redfirstletter::first-letter { background: red; }
+</style>
+<p>PASS if there is no red.</p>
+<ol>
+  <li>
+    <div class="redfirstline">
+      <math style="display: inline math"><mtext>Hello<br>World!</mtext></math>
```

Comment:
I'm not sure about HTML5 parsing rules exactly, but please check that in the DOM tree the HTML elements are children of  `<mtext>`. You may need to wrap in a span like this was done in 001.

Reply:
It seems to parse correctly in al 3 engines